### ISSUE TITLE
Question about masking input tokens

### ISSUE THREAD

**Issue by letsmakeadeal** [uvavision/RerankingTransformer#6]
Hello!
Can you explain please why are you masking global descriptors for pairs here RRT_GLD/models/matcher.py
```
        input_feats = torch.cat([cls_embed, src_global, src_local, sep_embed, tgt_global, tgt_local], 1).permute(1,0,2)
        input_mask = torch.cat([
            src_local.new_zeros((bsize, 2), dtype=torch.bool),
            src_mask,
            src_local.new_zeros((bsize, 2), dtype=torch.bool),
            tgt_mask
        ], 1)
        logits = self.encoder(input_feats, src_key_padding_mask=input_mask)
```
